diff --git a/node_modules/react-native-google-places-autocomplete/GooglePlacesAutocomplete.js b/node_modules/react-native-google-places-autocomplete/GooglePlacesAutocomplete.js
index 60836de..fa5fc10 100644
--- a/node_modules/react-native-google-places-autocomplete/GooglePlacesAutocomplete.js
+++ b/node_modules/react-native-google-places-autocomplete/GooglePlacesAutocomplete.js
@@ -275,7 +275,8 @@ export const GooglePlacesAutocomplete = forwardRef((props, ref) => {
       // fetch details
       const request = new XMLHttpRequest();
       _requests.push(request);
-      request.timeout = props.timeout;
+      request.timeout =
+        typeof props.timeout === 'number' ? props.timeout : 20000;
       request.ontimeout = props.onTimeout;
       request.onreadystatechange = () => {
         if (request.readyState !== 4) return;
@@ -471,7 +472,8 @@ export const GooglePlacesAutocomplete = forwardRef((props, ref) => {
     ) {
       const request = new XMLHttpRequest();
       _requests.push(request);
-      request.timeout = props.timeout;
+      request.timeout =
+        typeof props.timeout === 'number' ? props.timeout : 20000;
       request.ontimeout = props.onTimeout;
       request.onreadystatechange = () => {
         if (request.readyState !== 4) {
@@ -554,7 +556,8 @@ export const GooglePlacesAutocomplete = forwardRef((props, ref) => {
     if (supportedPlatform() && text && text.length >= props.minLength) {
       const request = new XMLHttpRequest();
       _requests.push(request);
-      request.timeout = props.timeout;
+      request.timeout =
+        typeof props.timeout === 'number' ? props.timeout : 20000;
       request.ontimeout = props.onTimeout;
       request.onreadystatechange = () => {
         if (request.readyState !== 4) {
@@ -904,7 +907,7 @@ export const GooglePlacesAutocomplete = forwardRef((props, ref) => {
     clearButtonMode,
     InputComp,
     ...userProps
-  } = props.textInputProps;
+  } = props.textInputProps || {};
   const TextInputComp = InputComp || TextInput;
   return (
     <View
